python/ray/llm/_internal/common/utils/download_utils.py

```diff
@@ -316,7 +333,195 @@ def download_lora_adapter(
     if remote_path is None:
         return lora_name
 
+    # Use the new unified LoRA downloading functionality
     lora_path = os.path.join(remote_path, lora_name)
+
+    # If advanced parameters are provided, use the new LoraModelLoader
+    if lora_root is not None or download_timeout_s is not None or max_tries != 1:
+        loader = _LoraModelLoader(
+            lora_root=lora_root,
+            download_timeout_s=download_timeout_s,
+            max_tries=max_tries,
+        )
+
+        # Create a simple LoraMirrorConfig for the loader
+        lora_mirror_config = LoraMirrorConfig(
+            lora_model_id=lora_name,
+            bucket_uri=lora_path,
+            max_total_tokens=None,
+        )
+
+        # Use the sync version directly
+        disk_config = loader._load_model_sync(lora_mirror_config)
+        return disk_config.local_path
+
+    # Fallback to original simple implementation for backward compatibility
     mirror_config = CloudMirrorConfig(bucket_uri=lora_path)
     downloader = CloudModelDownloader(lora_name, mirror_config)
     return downloader.get_model(tokenizer_only=False)
+
+
+class _LoraModelLoader:
```

Comment:
`download_lora_adapter` and `_LoRAModelLoader` need to be moved to `lora_utils.py`?